src/uct/tcp/tcp_ep.c

```diff
@@ -519,59 +509,55 @@ static void uct_tcp_ep_handle_disconnected(uct_tcp_ep_t *ep,
     }
 }
 
-static inline unsigned uct_tcp_ep_send(uct_tcp_ep_t *ep)
+static inline unsigned uct_tcp_ep_send(uct_tcp_ep_t *ep, size_t *sent_length)
 {
     uct_tcp_iface_t *iface = ucs_derived_of(ep->super.super.iface,
                                             uct_tcp_iface_t);
-    size_t send_length;
     ucs_status_t status;
 
-    send_length = ep->tx.length - ep->tx.offset;
-    ucs_assert(send_length > 0);
+    *sent_length = ep->tx.length - ep->tx.offset;
+    ucs_assert(*sent_length > 0);
 
     status = ucs_socket_send_nb(ep->fd, ep->tx.buf + ep->tx.offset,
-                                &send_length, NULL, NULL);
+                                sent_length, NULL, NULL);
     if (status != UCS_OK) {
         return 0;
     }
 
-    iface->outstanding -= send_length;
-    ep->tx.offset      += send_length;
-    ucs_trace_data("tcp_ep %p: sent %zu bytes", ep, send_length);
+    iface->outstanding -= *sent_length;
+    ep->tx.offset      += *sent_length;
 
-    return send_length > 0;
+    return (*sent_length > 0);
 }
 
-static inline unsigned uct_tcp_ep_sendv(uct_tcp_ep_t *ep)
+static inline unsigned uct_tcp_ep_sendv(uct_tcp_ep_t *ep, size_t *sent_length)
 {
     uct_tcp_iface_t *iface      = ucs_derived_of(ep->super.super.iface,
                                                  uct_tcp_iface_t);
     uct_tcp_ep_zcopy_ctx_t *ctx = (uct_tcp_ep_zcopy_ctx_t*)ep->tx.buf;
-    size_t send_length;
     ucs_status_t status;
 
     ucs_assertv(ep->tx.offset < ep->tx.length, "ep=%p", ep);
 
     status = ucs_socket_sendv_nb(ep->fd, &ctx->iov[ctx->iov_index],
                                  ctx->iov_cnt - ctx->iov_index,
-                                 &send_length, NULL, NULL);
-
-    ucs_trace_data("tcp_ep %p: sendv %zu bytes", ep, send_length);
+                                 sent_length, NULL, NULL);
 
-    ep->tx.offset      += send_length;
-    iface->outstanding -= send_length;
+    ep->tx.offset      += *sent_length;
+    iface->outstanding -= *sent_length;
```

Comment:
what is the reason to change send_tlength to *sent_length? is this a bugfix?

Reply:
not a bugfix, we should let caller known about how much data was sent to print it though `ucs_trace_data`

Reply:
per offline discussion:
will print the following info:
`"SEND: ep %p fd %d sent %zu/%zu bytes, moved to offest %zu, ..."`